tests/test_type/test_truncation.py

```diff
@@ -0,0 +1,19 @@
+import magma as m
+import pytest
+
+
+def test_python_int_truncation_error():
```

Comment:
let's just condense these into a single test and test the issue directly:

```python
def test_python_bits_from_int_truncation_error():
    with pytest.raises(ValueError) as e:
        m.Bits[2](4)
    assert str(e.value) == (
        "Cannot construct Array[2, Bit] with integer 4 (requires truncation)")
    with pytest.raises(ValueError) as e:
        m.bits(4, 2)
    assert str(e.value) == "Cannot convert 4 to a Bits of length 2"
```

Reply:
I was going to say it would be nice if there was just a single place we caught this -- in the `Array`/`Bits` constructor, but it seems like the way the `bits()` conversion function works, that won't be feasible. Might be worth considering consolidating these into the type constructors instead of external conversion functions.

Reply:
Yea I think there is actually the same/similar logic to do conversion inside the constructor, let's open an issue to try to unify where these conversions are done (probably makes the most sense to have it in the type constructor).